test/bioprinttest.c

```diff
@@ -244,11 +244,42 @@ static int test_fp(int i)
 static int test_big(void)
 {
     char buf[80];
+    double d, z, inf, nan;
 
     /* Test excessively big number. Should fail */
     if (!TEST_int_eq(BIO_snprintf(buf, sizeof(buf),
                                   "%f\n", 2 * (double)ULONG_MAX), -1))
         return 0;
+
+    d = 1.0;
+    z = 0.0;
+    inf = d / z;
+    nan = z / z;
```

Comment:
Would a sanity check: `TEST_false(nan == nan)` make sense?

It might be better in `sanitytest.c` and isn't all that relevant to cryptography (at least cryptography as it stands now).

Anyway, not in this PR.

Reply:
Yes, I have to agree it is not at all obvious for a crypto library to have a snprintf function...